core/domain/acl_decorators.py

```diff
@@ -123,9 +39,15 @@ def can_play_exploration(handler):
     """Decorator to check whether user can play given exploration."""
 
     def test_can_play(self, exploration_id, **kwargs):
-        if check_activity_accessible(
+        if exploration_id in feconf.DISABLED_EXPLORATION_IDS:
+            raise self.PageNotFoundException
+
+        exploration_rights = rights_manager.get_exploration_rights(
+            exploration_id, strict=False)
+
+        if rights_manager.check_can_access_activity(
                 self.user_id, self.actions, feconf.ACTIVITY_TYPE_EXPLORATION,
-                exploration_id):
+                exploration_rights):
```

Comment:
Just a thought: do you think it is worth returning the exploration_rights in the handler -- i.e. `return handler(self, exploration_id, exploration_rights, **kwargs)` -- in order to remove the need for a second retrieval? Might want to look at /_ah/stats and see how this affects the latency. Again it's fine to do this in a separate PR if you like.